fix bouncer workflow deployment

Fix error:

Run helm upgrade --install \
Pulled: ***/helm/bouncer:5.22.2
Digest: sha256:edf26ac8ec82fb7eebec521777fc06450bc0647737d6f6d60243ddbdd6d9b75d
Error: UPGRADE FAILED: template: bouncer/templates/deployment.yaml:35:22: executing "bouncer/templates/deployment.yaml" at <.Values.rollingupdate.enabled>: nil pointer evaluating interface {}.rollingupdate
Error: Process completed with exit code 1.

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -75,6 +75,7 @@ jobs:
             -f deployments/asite-stg/helm/bouncer/values.yaml \
             --set image.tag=${{ env.IMAGE_TAG }} \
             --set rollingupdate.enabled=true \
+            --set-string podAnnotations.buildId=${{ github.event.workflow_run.id }} \
             --wait \
             --rollback-on-failure
 

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -69,10 +69,6 @@ jobs:
       # ---------------------------------------------------------------
       # STEP 5: Deploy Helm chart
       # ---------------------------------------------------------------
-      - name: Debug values file
-        run: cat deployments/asite-stg/helm/bouncer/values.yaml || echo "FILE NOT FOUND"
-
-
       - name: Deploy Helm chart
         run: |
           helm upgrade --install \
@@ -82,7 +78,8 @@ jobs:
             --namespace default \
             -f deployments/asite-stg/helm/bouncer/values.yaml \
             --set image.tag=${{ env.IMAGE_TAG }} \
-            --set podAnnotations.buildId=${{ github.event.workflow_run.id }} \
+            --set rollingupdate.enabled=false \
+            --set-string podAnnotations.buildId=${{ github.event.workflow_run.id }} \
             --wait \
             --atomic
 

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -18,11 +18,11 @@ concurrency:
 
 env:
   HELM_CHART_VERSION: '5.22.2'
-  IMAGE_TAG: ${{ github.event.workflow_run.head_branch || 'staging' }}
+  IMAGE_TAG: ${{ github.event.workflow_run.head_branch }}
 
 jobs:
   deploy:
-    if: github.event_name == 'push' || github.event.workflow_run.conclusion == 'success'
+    if: github.event.workflow_run.conclusion == 'success'
     runs-on: ubuntu-latest
 
     steps:
@@ -75,7 +75,8 @@ jobs:
             -f deployments/asite-stg/helm/bouncer/values.yaml \
             --set image.tag=${{ env.IMAGE_TAG }} \
             --set rollingupdate.enabled=true \
-            --wait
+            --wait \
+            --rollback-on-failure
 
       # ---------------------------------------------------------------
       # STEP 6: Cleanup

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -78,7 +78,7 @@ jobs:
             --namespace default \
             -f deployments/asite-stg/helm/bouncer/values.yaml \
             --set image.tag=${{ env.IMAGE_TAG }} \
-            --set-string podAnnotations.buildId=${{ github.event.workflow_run.id }} \
+            --set-string podAnnotations.buildId=${{ github.event.workflow_run.id || github.run_id }} \
             --wait
 
       # ---------------------------------------------------------------

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -69,6 +69,10 @@ jobs:
       # ---------------------------------------------------------------
       # STEP 5: Deploy Helm chart
       # ---------------------------------------------------------------
+      - name: Debug values file
+        run: cat deployments/asite-stg/helm/bouncer/values.yaml || echo "FILE NOT FOUND"
+
+
       - name: Deploy Helm chart
         run: |
           helm upgrade --install \

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -22,11 +22,11 @@ concurrency:
 
 env:
   HELM_CHART_VERSION: '5.22.2'
-  IMAGE_TAG: ${{ github.event.workflow_run.head_branch }}
+  IMAGE_TAG: ${{ github.event.workflow_run.head_branch || 'staging' }}
 
 jobs:
   deploy:
-    if: github.event.workflow_run.conclusion == 'success'
+    if: github.event_name == 'push' || github.event.workflow_run.conclusion == 'success'
     runs-on: ubuntu-latest
 
     steps:

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -78,7 +78,7 @@ jobs:
             --namespace default \
             -f deployments/asite-stg/helm/bouncer/values.yaml \
             --set image.tag=${{ env.IMAGE_TAG }} \
-            --set-string podAnnotations.buildId=${{ github.event.workflow_run.id || github.run_id }} \
+            --set rollingupdate.enabled=true \
             --wait
 
       # ---------------------------------------------------------------

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -78,10 +78,8 @@ jobs:
             --namespace default \
             -f deployments/asite-stg/helm/bouncer/values.yaml \
             --set image.tag=${{ env.IMAGE_TAG }} \
-            --set rollingupdate.enabled=false \
             --set-string podAnnotations.buildId=${{ github.event.workflow_run.id }} \
-            --wait \
-            --atomic
+            --wait
 
       # ---------------------------------------------------------------
       # STEP 6: Cleanup

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -1,6 +1,10 @@
 name: Deploy Bouncer to Staging
 
 on:
+  push:
+    branches:
+      - ISSUE_867
+
   workflow_run:
     workflows: ["Build and Push Bouncer"]
     types: [completed]
@@ -43,7 +47,7 @@ jobs:
       - name: Setup Helm
         uses: azure/setup-helm@v4
         with:
-          version: '3.10.3'
+          version: '4.0.1'
 
       # ---------------------------------------------------------------
       # STEP 3: Set Kubernetes context

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -1,10 +1,6 @@
 name: Deploy Bouncer to Staging
 
 on:
-  push:
-    branches:
-      - ISSUE_867
-
   workflow_run:
     workflows: ["Build and Push Bouncer"]
     types: [completed]

